dvc/project.py

```diff
@@ -412,7 +404,8 @@ def imp(self, url, out):
         if stage is None:
             return None
 
-        self._check_output_duplication(stage.outs, self.stages())
+        # Generate graph including the new stage to check for errors
+        self.graph(stages=self.stages() + [stage])
```

Comment:
Imported stage might also be overwritten. Maybe it is worth simply removing stage file in stage.py:create() when overwrite==True? What do you think?